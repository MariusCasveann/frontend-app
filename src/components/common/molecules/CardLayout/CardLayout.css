.card-layout {
    display: flex;
    display: -ms-flexbox;
    flex-wrap: wrap;
    justify-content: space-between;
    background-color: transparent;
}

.card {
    border-bottom: 1px solid transparent;
    margin: 10px;
    padding: 15px;
    background-color: white;
    box-shadow: 0 0 6px 2px rgba(0, 0, 0, 0.1);
    transition: all 0.2s ease;
    min-width: 320px;
}

.card:hover {
    border-bottom: 1px solid #25a7f7;
}

.card-title-wrapper {
    display: flex;
    justify-content: space-between;
    padding: 5px;
    box-shadow: 0 3px 0px rgba(0, 0, 0, 0.04), 0 3px 0px rgba(0, 0, 0, 0.04);
}

.card-content-data {
    display: flex;
    display: -ms-flexbox;
    align-items: stretch;
    justify-content: stretch;
    min-height: 50px;
}

.card-content-wrapper {
    flex: 1;
    display: block;
    padding-top: 20px;
    padding: 20px 10px 0px 10px;
}

.card-content-loading,
.card-content-error {
    flex: 1;
    display: flex;
    display: -ms-flexbox;
    flex-direction: column;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    min-height: 50px;
    padding: 20px;
}

.card-title {
    font-family: sans-serif;
    font-size: 16px;
    font-weight: 700;
    color: var(--font-color-copy);
}

.card-selection {
    text-align: right;
    width: 60%;
}

/* Style for mobile phone */
@media only screen and (max-width: 576px) {
    .card .flex-50 {
        flex: 1 1 calc(100% - 10px);
    }

    .card-title {
        font-family: sans-serif;
        font-size: 14px;
        font-weight: 700;
    }

    .card {
        min-width: 300px;
    }
}

/* Style for tablet */
@media only screen and (max-width: 768px) {
    .flex-50 {
        flex: 1 1 calc(50% - 15px);
    }

    .card-title {
        font-family: sans-serif;
        font-size: 14px;
        font-weight: 700;
    }
}
