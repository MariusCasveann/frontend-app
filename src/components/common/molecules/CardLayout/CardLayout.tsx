import classNames from 'classnames';
import React, { ReactNode } from 'react';
import { FaExclamationCircle } from 'react-icons/fa';
import { Spin } from 'antd';
import './CardLayout.css';

interface CardProps {
    cardTitleClass?: string;
    children: ReactNode;
    className?: string;
    error?: boolean;
    loading?: boolean;
    renderSelection?: () => ReactNode;
    selectionWidth?: number;
    title?: string;
    titleWidth?: number;
    width: number;
}

const CardTitle = ({ title, renderSelection, cardTitleClass }: Partial<CardProps>) => (
    <div className="card-title-wrapper">
        {title ? <div className="card-title">{title}</div> : undefined}
        {renderSelection ? <div className={`card-selection ${cardTitleClass}`}>{renderSelection()}</div> : undefined}
    </div>
);

export const Card = ({
    className,
    cardTitleClass,
    width,
    title,
    titleWidth,
    selectionWidth,
    renderSelection,
    children,
    loading,
    error
}: CardProps) => (
    <div className={classNames(className, 'card')} style={{ flex: `1 1 ${width}%` }}>
        {title || renderSelection ? (
            <CardTitle
                title={title}
                renderSelection={renderSelection}
                titleWidth={titleWidth}
                selectionWidth={selectionWidth}
                cardTitleClass={cardTitleClass}
            />
        ) : (
            undefined
        )}
        {error ? (
            <div className={classNames('card-content-error')}>
                <div>
                    <FaExclamationCircle size={48} />
                </div>
                <div>An error has occured. Please contact an administrator</div>
            </div>
        ) : !loading ? (
            <div className={classNames('card-content-data')}>
                <div className="card-content-wrapper">{children}</div>
            </div>
        ) : (
            <div className="card-content-loading">
                <Spin size="large" spinning={true} />
            </div>
        )}
    </div>
);

interface CardLayoutProps {
    className?: string;
    width: number;
    children: ReactNode;
}

export const CardLayout = ({ className, width, children }: CardLayoutProps) => (
    <div className={classNames('card-layout', className)} style={{ flex: `1 1 ${width}%` }}>
        {children}
    </div>
);
