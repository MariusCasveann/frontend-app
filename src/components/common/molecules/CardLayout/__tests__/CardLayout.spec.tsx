import * as React from 'react';
import { CardLayout, Card } from './../CardLayout';
import renderer from 'react-test-renderer';

const mockWidth = 30;
const mockChildren = <div>Hello</div>;

describe('Card', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Card children={mockChildren} width={mockWidth} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Card', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <CardLayout width={mockWidth}>
                    <Card children={mockChildren} width={mockWidth} />
                </CardLayout>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
