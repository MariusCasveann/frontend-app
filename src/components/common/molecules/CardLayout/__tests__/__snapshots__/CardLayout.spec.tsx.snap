// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "flex": "1 1 30%",
    }
  }
>
  <div
    className="card-content-data"
  >
    <div
      className="card-content-wrapper"
    >
      <div>
        Hello
      </div>
    </div>
  </div>
</div>
`;

exports[`Card renders correctly 2`] = `
<div
  className="card-layout"
  style={
    Object {
      "flex": "1 1 30%",
    }
  }
>
  <div
    className="card"
    style={
      Object {
        "flex": "1 1 30%",
      }
    }
  >
    <div
      className="card-content-data"
    >
      <div
        className="card-content-wrapper"
      >
        <div>
          Hello
        </div>
      </div>
    </div>
  </div>
</div>
`;
